modules/core/src/main/java/org/apache/ignite/internal/processors/cache/IgniteCacheOffheapManagerImpl.java

```diff
@@ -1900,10 +1900,12 @@ private boolean isIncrementalDrEnabled(GridCacheContext cctx) {
             else {
                 CacheObjectContext coCtx = cctx.cacheObjectContext();
 
-                key.prepareForCache(coCtx, coCtx.compressKeys());
+                key = key.prepareForCache(coCtx, coCtx.compressKeys());
 
                 if (val != null)
-                    val.prepareForCache(coCtx, true);
+                    val = val.prepareForCache(coCtx, true);
+
+                dataRow = makeDataRow(key, val, ver, expireTime, cacheId);
```

Comment:
Well, it looks like we always create a new `dataRow` even though `key` and `val` are not changed after `prepareForCache`. Is it OK?

Reply:
Why are they not changed? We reassign both key and val, and prepareForCache() may return a new object.

Reply:
canUpdateOldRow() guarantees that compression is off.

Reply:
Well, do we really need to create a new data row in case of entry compression is disabled?